src/libaudcore/mainloop.cc

```diff
@@ -224,13 +224,27 @@ class QueuedFuncTimer : public QueuedFuncHelper, public QObject
     }
 
     void stop ()
-        { deleteLater (); }
+    {
+        if (timerId)
+        {
+            killTimer(timerId);
+            timerId = 0;
+        }
+        deleteLater ();
+    }
 
 protected:
     void customEvent (QEvent *)
-        { startTimer (params.interval_ms); }
-    void timerEvent (QTimerEvent *)
-        { run (); }
+        { timerId = startTimer (params.interval_ms); }
+    void timerEvent (QTimerEvent * event)
+    {
+        if (event->timerId() == timerId)
```

Comment:
This check shouldn't be necessary; the object should only ever have one timer running.

Reply:
Not merging any of the changes in this file unless they're clearly shown to be necessary.